### DISCUSSION TITLE
How to create Pull Request in Github repo?

### DISCUSSION THREAD

**Discussion by fullstackdev48** [fullstackdev48/shopify-template-ruby#10] [Q&A] [answered]
How to create Pull Request in Github repo?

**Comment by baby1194** [comment, marked as answer]
Creating a pull request (PR) in a GitHub repository involves a few steps. Here's a general guide:

Fork the Repository: Go to the GitHub repository you want to contribute to and click on the "Fork" button in the top-right corner. This will create a copy of the repository in your GitHub account.
Clone the Forked Repository: After forking, clone the repository to your local machine using Git. You can do this by running the following command in your terminal:
git clone <repository_url>

**Comment by outdoorSpirit** [comment]
Creating a Pull Request (PR) on GitHub involves several steps. Below is a step-by-step guide:

### Step 1: Fork the Repository

If you do not have write access to the repository, you'll need to create a fork.

1. Navigate to the GitHub repository you want to contribute to.
2. In the top-right corner of the page, click the "Fork" button.

This will create a copy of the repository in your GitHub account.

### Step 2: Clone Your Forked Repository

Clone the repository to your local machine:

```sh
git clone https://github.com/YOUR-USERNAME/REPOSITORY-NAME.git
cd REPOSITORY-NAME
```

Replace `YOUR-USERNAME` and `REPOSITORY-NAME` with your GitHub username and the name of the repository.

### Step 3: Create a New Branch

Create a new branch for your changes:

```sh
git checkout -b feature-branch-name
```

Replace `feature-branch-name` with a descriptive name for your branch.

### Step 4: Make Changes and Commit Them

Make the necessary changes in your local repository. After making the changes, stage and commit them:

```sh
git add .
git commit -m "Your detailed description of your changes"
```

### Step 5: Push the Changes to Your Fork

Push the changes to your fork on GitHub:

```sh
git push origin feature-branch-name
```

### Step 6: Create the Pull Request

1. Go back to the original repository on GitHub.
2. You'll likely see a "Compare & pull request" button for your recently pushed branch. Click this button.
3. If not, switch to the branch using the branch dropdown menu, and hit the "New pull request" button.
4. Make sure the base fork is set to the repository you originally forked from, and the base branch is the one you want to merge your changes into.
5. The head fork should be your fork, and the compare branch should be the feature branch you created.
6. Fill in a title that describes your changes, and provide a detailed description in the comment box. This is important as it helps maintainers understand what you've done and why.
7. If the repository has a template for pull requests, make sure to fill it out.
8. Once ready, click "Create pull request".

After submitting, the project maintainers will review your PR. They may suggest changes, which you can make by pushing more commits to the same branch. When the PR is approved, the maintainers will merge your changes into the repository.
